File: 7.Reverse Integer/sr333444/ReverseInteger.c
/*4ms*/
int reverse(int x) {
    int result = 0;
	while(true)
	{
		if(result>0 && result>(INT_MAX-x%10)/10)
		{
			result = 0;
			break;
		}
		else if(result<0 && result<(INT_MIN-x%10)/10)
		{
			result=0;
			break;
		}
		result = result * 10 + x % 10; 
		x /= 10;
		
		if(x == 0)
			break;
	}
	return result;
}

File: 007.Reverse Integer/Ddfish/ReverseInteger.swift
class Solution {
    func reverse(_ x: Int) -> Int {
        
        var xInt32 : Int32 = Int32(x)
        var answer : Int32 = 0
        
        while xInt32 != 0 {
            
            let result = Int32.multiplyWithOverflow(answer, Int32(10))
            let digit = xInt32 % 10
            xInt32 /= 10
            if(result.overflow){
                return 0
            }else{
                answer = result.0 + digit
            }
        }
        
        return Int(answer)

    }
}
